node/packages/parser/src/ethereum/thor.ts

```diff
@@ -48,21 +49,28 @@ export class Parser {
     return result.to
   }
 
+  isSupportedTransaction(tx: Tx) {
+    return txMatchesContract(tx, this.routerContract)
+  }
+
   async parse(tx: Tx): Promise<ParseTxSpecific | undefined> {
-    if (!tx.ethereumSpecific?.data) return
-
-    let result: ethers.utils.Result
-    switch (getSigHash(tx.ethereumSpecific.data)) {
-      case this.depositSigHash:
-        result = this.abiInterface.decodeFunctionData(this.depositSigHash, tx.ethereumSpecific.data)
-        break
-      case this.transferOutSigHash: {
-        result = this.abiInterface.decodeFunctionData(this.transferOutSigHash, tx.ethereumSpecific.data)
-        break
+    if (!this.isSupportedTransaction(tx)) return
+    if (!(isEthereumSpecific(tx.ethereumSpecific) && tx.ethereumSpecific.data)) return
+
+    const result: ethers.utils.Result | undefined = (() => {
```

Comment:
Down to make use of anon inline functions to avoid `let`. Is the type definition required, or is the implicit type assigned by default?

Reply:
Implicitly assigned - removed in https://github.com/shapeshift/unchained/pull/238/commits/143f663f3aa13c278ce57bef914db30cc1be0599.